src/trie/trie.c

```diff
@@ -80,21 +80,48 @@ TrieNode *__newTrieNode(const rune *str, t_len offset, t_len len, const char *pa
   return n;
 }
 
+TrieNode *__trieNode_resizeChildren(TrieNode *n, int offset) {
+  n = rm_realloc(n, __trieNode_Sizeof(n->numChildren + offset, n->len));
+  TrieNode **children = __trieNode_children(n);
+
+  // stretch or shrink the child key cache array
+  memmove(((rune *)children) + offset, (rune *)children, sizeof(TrieNode *) * n->numChildren);
+  n->numChildren += offset;
+  return n;
+}
+
 TrieNode *__trie_AddChild(TrieNode *n, const rune *str, t_len offset, t_len len, RSPayload *payload,
                           float score) {
-  n->numChildren++;
-  n = rm_realloc((void *)n, __trieNode_Sizeof(n->numChildren, n->len));
+  n = __trieNode_resizeChildren(n, 1);                            
   // a newly added child must be a terminal node
   TrieNode *child = __newTrieNode(str, offset, len, payload ? payload->data : NULL,
                                   payload ? payload->len : 0, 0, score, 1, n->sortMode);
   __trieNode_children(n)[n->numChildren - 1] = child;
-
+  *__trieNode_childKey(n, n->numChildren - 1) = str[offset];
   // Sort children on write to avoid sorting on reads
   __trieNode_sortChildren(n);
```

Comment:
Why you still need the sorting here if you are doing memmove? One of them are not needed no?

Reply:
Remove function completely.
Only remove by index variant left.